src/vimscript/pattern.ts

```diff
@@ -83,8 +84,16 @@ export class Pattern {
       fromPosition = args.fromPosition;
     }
 
-    const haystack = vimState.document.getText();
-    const startOffset = vimState.document.offsetAt(fromPosition);
+    const visualSearch = this.inselection && vimState.lastVisualSelection;
+    const haystack = vimState.document.getText(
+      visualSearch
+        ? new Selection(vimState.lastVisualSelection!.start, vimState.lastVisualSelection!.end)
```

Comment:
No need to import `Selection` - you can make this `new Selection(...)`

Reply:
fixed